core/src/main/java/io/aiven/klaw/helpers/db/rdbms/SelectDataJdbc.java

```diff
@@ -1169,4 +1156,57 @@ public int findAllComponentsCountForTeam(Integer teamId, int tenantId) {
   public int getAllTopicsCountInAllTenants() {
     return ((Long) topicRepo.findAllTopicsCount().get(0)[0]).intValue();
   }
+
+  // teamId is requestedBy. For 'topics' all the requests are assigned to the same team, except
+  // claim requests
+  public Map<String, Map<String, Long>> getTopicRequestsCounts(
+      int teamId, RequestMode requestMode, int tenantId) {
+    Map<String, Map<String, Long>> allCountsMap = new HashMap<>();
+
+    Map<String, Long> operationTypeCountsMap = new HashMap<>();
+    Map<String, Long> statusCountsMap = new HashMap<>();
+
+    if (RequestMode.MY_REQUESTS == requestMode) {
+      List<Object[]> topicRequestsOperationTypObj =
+          topicRequestsRepo.findAllTopicRequestsGroupByOperationType(teamId, tenantId);
+      topicRequestsOperationTypObj.forEach(
+          topicReqObjs ->
+              operationTypeCountsMap.put((String) topicReqObjs[0], (Long) topicReqObjs[1]));
+
```

Comment:
NitPick: This operation as well could probably be pulled into a single method and reused.

Reply:
@aindriu-aiven updated. good one.